xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/edit.vm

```diff
@@ -100,6 +100,13 @@
     <div class="hidden">
     ## CSRF prevention
     <input type="hidden" name="form_token" value="$!{services.csrf.getToken()}" />
+    ## Current version of the document for the conflict check
+    <input type="hidden" id="previousVersion" name="previousVersion" value="$doc.version" />
+    ## If the document is new or not when the user started the edition, for the conflict check too.
+    <input type="hidden" id="isNew" name="isNew" value="$doc.isNew()" />
+    ## We store the date of the beginning of the edition for the conflict check mechanism.
+    <input type="hidden" id="editingVersionDate" name="editingVersionDate" value="$datetool.date.getTime()" />
```

Comment:
What about the places where ``actionButtons.js`` is used outside of edit mode:
* preview
* administration sections
* repository application
* in-place editing

If these hidden fields are used by ``actionButtons.js``, why not generate them in the same place where the action buttons are generated, i.e. ``editactions.vm``?

Reply:
> why not generate them in the same place where the action buttons are generated, i.e. editactions.vm?

So I use to have a special check in actionButtons.js to not generate those fields outside an /edit/ action (see: https://github.com/xwiki/xwiki-platform/pull/1409/files#diff-c96ca895f6b4766e80a3282f4b68203aL247), that's why I put it here. 
Now for sure I should also support it on preview and in-place editing. I will check to put them directly in editaction.vm, but to be honest I'm a bit afraid about the impact on AWM and Administration. Now a solution could be to do the contrary than https://github.com/xwiki/xwiki-platform/pull/1409/files#diff-c96ca895f6b4766e80a3282f4b68203aL247 i.e. to remove the fields if I'm not in edit.

Reply:
Actually from what I see for preview we don't use editactions.vm but previewactions.vm. Sounds like I will have to multiply those fields in all templates where I want to cache the info.

Reply:
> Actually from what I see for preview we don't use editactions.vm but previewactions.vm. Sounds like I will have to multiply those fields in all templates where I want to cache the info.

Apparently all editors (including inline) uses editactions.vm, so I'd need to put them as mentioned also in `previewactions.vm`, now IMO it makes sense to not put it for other cases you mentioned (Administration / Repository application). Do you agree?

Reply:
Actually I shouldn't even need to put them in `previewactions.vm` since the parameters are automatically added as hidden field for preview. So should be ok only with `editactions.vm` for that.